app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebasePlantsRepository.kt

```diff
@@ -77,8 +106,51 @@ class FirebasePlantsRepository @Inject constructor(
         return plantsCollection.document(plantName.value)
     }
 
+    private suspend fun addImageToStorage(picture: Uri): Uri {
+        val storageImagePath = STORAGE_PICTURES_DIR_PATH + picture.lastPathSegment
+        storageRef
+            .child(storageImagePath)
+            .apply {
+                putFile(picture).await()
+            }
+
+        return storageImagePath.toUri()
+    }
+
+    private suspend fun saveImageToLocalStorage(cloudPicturePath: String): Uri? {
+        return when {
+            cloudPicturePath.isNotBlank() -> {
+                val imageRef = storageRef.child(cloudPicturePath)
+
+                val localFile = File(
+                    getLocalStorageImagePath(imageRef.name)
+                )
+
+                if (!localFile.exists()) {
+                    imageRef.getFile(localFile).await()
+                }
+
+                context.getFileUri(localFile)
+            }
+            else -> null
+        }
+    }
```

Comment:
Have a question about it: is it nesessary to do so? Or maybe we can just use some url?